internal/store/assessment.go

```diff
@@ -105,12 +126,8 @@ func (a *AssessmentStore) Update(ctx context.Context, assessmentID uuid.UUID, na
 	}
 
 	if inventory != nil {
-		// Create a new snapshot
-		snapshot := model.Snapshot{
-			AssessmentID: assessmentID,
-			Inventory:    model.MakeJSONField(*inventory),
-		}
-
+		// Create a new snapshot using the helper function
+		snapshot := a.buildInitialSnapshot(assessmentID, inventory)
```

Comment:
this is trickier. The user could update only the name of the assessment, and in this case, there is no inventory, and no snapshot is created. 
I think you need to split this into two: UpdateName UpdateInventory, to be able to differentiate between the usecases.

Reply:
No. he should be able to update assessment's name without creating a new snapshot